### PR TITLE
Bugfix: It breaks events and animations at every run

### PR DESCRIPTION
**Pull request by bfred-it**
#63

No support for justified text yet, which I'm not sure if it should be there at all, it should be part of a `text-align-last` polyfill.

### CODE REVIEW

**Comment by redmunds** [comment]
@bfred-it I'm trying to find some time to review this. You'll need to merge with the latest master branch to get the change I just merge from your [other PR](#67).

Please provide a sample page that illustrates this problem. If it's a small test, then I'd like to merge it into *test/test1.htm*, otherwise probably best to make it *test/test2.htm*.

As far as justified text, you're correct that it is orthogonal to text balancing and will be separate once this (hopefully) gets into the [CSS Text 4 module as `text-wrap:balance`](http://www.w3.org/TR/css-text-4/#text-wrap). But, since this polyfill runs after page load, justification doesn't get done on balanced text, so that's why I accepted that contribution. I'll look at fixing that in your branch.

**Comment by bfred-it** [comment]
Here we go. This is what I'm currently using on toyotahalloffame.com in a lot of places.

**Comment by redmunds** [comment]
@bfred-it I finally got a chance to look at this. My tests seem to work correctly.

I'm trying to get my head around the pros and cons of this change. The sample page you posted is very cool, but it's it's quite complex, so I have not yet dissected it. I created a simple animation that is applied with CSS and it still works after resizing page with existing code. Are you applying the animations in JS? If so, they could probably be re-applied in an onresize handler, right?

**Comment by bfred-it** [comment]
They cannot be reapplied correctly because the animation is stopped wherever that happens and if I reapply it I will have lost the initial position. Even if it did, that wouldn't be ideal. Each feature of the site would have to attach to the resize handler and attach them again *after* BT and animations would have to be reinitiated.

**Comment by bfred-it** [comment]
Ping :)

**Comment by redmunds** [comment]
@bfred-it Sorry, for not replying sooner. I am not sure if I want to take this change. It adds complexity and possibly impacts performance (neither of which I have had time to sufficiently test). This seems like a good way to fix this problem, but you're the only person to hit this so far. Also, this fix doesn't handle the justified text case.

The beauty of open source code is that you can have a fork of this repo and the community can have the choice of either solution. One downside is that some fixes in this repo may conflict with your changes and may not be easy to pull into your repo.

I am open to hear the opinions of everyone in the community and maybe I'll change my mind on this, but that's where I'm currently at. How does that sound?

**Comment by bfred-it** [comment]
I already have my own fork but I'm not sure I want to keep merging changes from here. The point of PRs is to make this available to everyone without having to _pull a lodash._

Performance should be improved since only some nodes are inserted/removed as opposed to entire node contents via `innerHTML`. If that helps, I can run some performance tests and report back the plots. My plan was to add the last-align feature back if there was the intention of pulling it, otherwise it wouldn't be useful to me.

**Comment by bfred-it** [comment]
And no worries about the time, we all have side-projects that can't get too much attention. =)

**Comment by bfred-it** [comment]
Here's a demo page showing all the issues that the current version has: http://codepen.io/bfred-it/pen/EyXZGK

File: jquery.balancetext.js [hunk 1]
```diff
@@ -384,13 +441,12 @@
...
-                    // Break string
-                    lineText = remainingText.substr(0, splitIndex);
                     if (shouldJustify) {
+                        // TODO, UPDATE. CURRENTLY BROKEN
                         newText += justify($this, lineText, containerWidth);
                     } else {
-                        newText += lineText.replace(/\s$/, "");
-                        newText += '<br data-owner="balance-text" />';
+                        usedTextIndex += splitIndex;
+                        breakIndexes.push(usedTextIndex)
```
Comment (redmunds) [inline jquery.balancetext.js]: Missing `;`

**Comment by redmunds** [comment]
@bfred-it This looks good. I'd like to see the codepen sample put in a test2.htm file. Also, did you get a chance to look at fixing justification?

**Pull request closed without merging**